### ISSUE TITLE
Videos not playing at all in Amazon VOD, problem with inputstreamaddon

### ISSUE THREAD

**Issue by HeuristicPerson** [Sandmann79/xbmc#543]
**Addon used**
- [X] Amazon VOD (plugin.video.amazon-test)
- [ ] Browser Launcher (plugin.program.browser.launcher)

- Addon version: 

**Account type**
- [ ] primevideo.com
- [X] amazon.(com/co.uk/de/jp)

**System Setup (please provide the following information):**
- Hardware:
- OS version: Linux Mint 19 Tara
- Kodi version number: 19.0 (19.0.0) Git:20210219-f44fdfbf67 Media Center Kodi

**Upload Logs**

    2021-02-21 14:04:11.378 T:13716 WARNING <general>: [script.module.inputstreamhelper] Widevine update check was made on 2021-02-21 13:43
    2021-02-21 14:04:11.386 T:13716 WARNING <general>: [Amazon VOD] postURL: http://127.0.0.1:46759/gpr/https%3A%2F%2Fatv-ps-eu.amazon.co.uk%2Fcdp%2Fcatalog%2FGetPlaybackResources%3Fasin%3DB086VXPS7T%26deviceTypeID%3DAOAGZA014O5RE%26firmware%3D1%26deviceID%3Dc623ab650adabd379514547ad5d000677ef5e5c3d10e3388c8c15842%26marketplaceID%3DA1F83G8C2ARO7P%26format%3Djson%26version%3D1%26gascEnabled%3Dfalse%26subtitleFormat%3DTTMLv2%26resourceUsage%3DImmediateConsumption%26consumptionType%3DStreaming%26deviceDrmOverride%3DCENC%26deviceStreamingTechnologyOverride%3DDASH%26deviceProtocolOverride%3DHttps%26deviceBitrateAdaptationsOverride%3DCVBR%252CCBR%26audioTrackId%3Dall%26languageFeature%3DMLFv2%26videoMaterialType%3DFeature%26desiredResources%3DPlaybackUrls%2CSubtitleUrls%2CForcedNarratives%2CTransitionTimecodes%26supportedDRMKeyScheme%3DDUAL_KEY
    2021-02-21 14:04:11.680 T:13716 WARNING <general>: [Amazon VOD] Using Host: Cloudfront
    2021-02-21 14:04:11.681 T:13716 WARNING <general>: [Amazon VOD] checkURL: https://pop-dub-ww.cf.dash.row.aiv-cdn.net/dm/2$waALjF6-gHAUT5QEyD6z4nAF7Kk~/8dea/c72c/3c50/460c-a868-8e1f7967045a/1984fb1e-3a52-4e92-9914-ed11d4272439_corrected.mpd
    2021-02-21 14:04:12.529 T:13716 WARNING <general>: [Amazon VOD] Using inputstream.adaptive Version: 2.6.7
    2021-02-21 14:04:12.585 T:28778    INFO <general>: VideoPlayer::OpenFile: plugin://plugin.video.amazon-test/?asin=B086VXPS7T&mode=PlayVideo&name=1+-+The+Holy+Trinity&adult=0&trailer=0&selbitrate=0
    2021-02-21 14:04:12.589 T:13731    INFO <general>: Creating InputStream
    2021-02-21 14:04:12.589 T:13731   ERROR <general>: CInputStreamAddon::Supports - 'inputstreamaddon' has been deprecated, please use `#KODIPROP:inputstream=inputstream.adaptive` instead
    2021-02-21 14:04:12.589 T:13731   ERROR <general>: IAddonInstanceHandler::CreateInstance: inputstream.adaptive returned bad status "Permanent failure" during instance creation
    2021-02-21 14:04:12.590 T:13731   ERROR <general>: CVideoPlayer::OpenInputStream - error opening [plugin://plugin.video.amazon-test/?asin=B086VXPS7T&mode=PlayVideo&name=1+-+The+Holy+Trinity&adult=0&trailer=0&selbitrate=0]
    2021-02-21 14:04:12.590 T:13731    INFO <general>: CVideoPlayer::OnExit()
    2021-02-21 14:04:12.701 T:28778    INFO <general>: CVideoPlayer::CloseFile()
    2021-02-21 14:04:12.701 T:28778    INFO <general>: VideoPlayer: waiting for threads to exit
    2021-02-21 14:04:12.702 T:28778    INFO <general>: VideoPlayer: finished waiting

**Describe the bug**

Videos don't play at all.

**Comment by Varstahl** [comment]
InputStream.Adaptive is being migrated to be fully under Kodi's development team. The changes have been implemented recently, I guess we'll still wait a little bit before breaking support for previous versions, but definitely on the todo list.

**Comment by F1nnM** [comment]
Hi, are there any estimates on when it will work again? Weeks, months, years?

**Comment by Varstahl** [comment]
TL;DR: revert back to a previous version of Kodi which still works, while you wait for an update.

Long version: COVID and full lockdown has done a number on me, and I haven't been able to as little as look at AVOD's code in months. Due to problematic real life circumstances I'm unable to provide even the faintest guesstimate.

The addon still works fine, but you will probably need to downgrade to a less bleeding edge release for the time being.

**Comment by F1nnM** [comment]
Oh, I'm sorry to hear that. That's very much understandable; please don't feel pressured by this project in those challenging times! I wish you all the best!

I've tried the most recent version as of yesterday, so that should be 0.9.2.

I'll just revert to an older KODI version then.

**Comment by HeuristicPerson** [comment]
Good afternoon, addon started to work again for me. I assume it's been an update on InputStream.

**Comment by kalyan-1986** [comment]
Hi, I am from india, 

I followed your kodi configure completely and installed the plugin, it's installed, logging also successful, but after that, if I opened any video, video is playing but the video is not visible and audio also not audible, please help me out.

**Comment by Varstahl** [comment]
@kalyan-1986 seems like a device playback issue, can't really help you there I'm afraid. If you have full logs open a new issue and we'll look into it, but if the video actually starts, I think it's beyond what this addon does.

Also closing this for self resolution.
